File: 1013-fibonacci-number/fibonacci-number.cpp
class Solution {
public:
    int arr[31];
    bool check[31];


    int fib(int n) {
        if (check[n]) return arr[n];
        if (n == 0) return 0;
        if (n <= 2) return 1;

        return arr[n] = fib(n-1) + fib(n-2);    
    }
};

File: 1013-fibonacci-number/fibonacci-number.go
func fib(n int) int {
    if n == 0 || n == 1 {
        return n
    }
    f, s := 0, 1
    for i := 2; i <= n; i++ {
        tmp := f + s
        f = s
        s = tmp
    }
    return s
}